Quaver.Shared/Screens/Selection/SelectionScreen.cs

```diff
@@ -574,6 +582,24 @@ private void ChangeScrollSpeed()
             }
         }
 
+        /// <summary>
+        ///     Undoes the random map choice
+        /// </summary>
+        public void SelectPrevRandomMap()
+        {
+            if (AvailableMapsets.Value.Count == 0)
+                return;
+
+            if (RngHistory.Count == 0)
+                return;
+
+            var map = RngHistory.Pop();
+
+            MapManager.Selected.Value = map;
```

Comment:
This needs to handle the case of when the user deletes the previous map in the history and then selects it.

Reply:
Now it will try all the way down to the history until it find an existing one(Fun fact: At the present time, Osu! has this same bug)